staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -207,10 +207,14 @@ func conditionFuncFor(condition string, errOut io.Writer) (ConditionFunc, error)
 	}
 	if strings.HasPrefix(condition, "jsonpath=") {
 		splitStr := strings.Split(condition, "=")
-		if len(splitStr) != 3 {
-			return nil, fmt.Errorf("jsonpath wait format must be --for=jsonpath='{.status.readyReplicas}'=3")
+		if len(splitStr) < 2 || len(splitStr) > 3 {
+			return nil, fmt.Errorf("jsonpath wait format must be --for=jsonpath='{.status.readyReplicas}'[=3]")
 		}
-		jsonPathExp, jsonPathCond, err := processJSONPathInput(splitStr[1], splitStr[2])
+		conditionStr := ""
+		if len(splitStr) == 3 {
+			conditionStr = splitStr[2]
+		}
+		jsonPathExp, jsonPathCond, err := processJSONPathInput(splitStr[1], conditionStr)
```

Comment:
nit: refactoring it by explicitly handling a case without "JSONPath condition" will reduce cognitive load when reading the code. I suggest to handle the cases `len(splitStr) == 2` and `len(splitStr) == 3` both for readability and code cleaness

Reply:
Done